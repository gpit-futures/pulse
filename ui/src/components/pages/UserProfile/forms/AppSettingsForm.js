import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
