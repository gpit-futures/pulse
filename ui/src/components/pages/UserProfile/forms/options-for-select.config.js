const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
