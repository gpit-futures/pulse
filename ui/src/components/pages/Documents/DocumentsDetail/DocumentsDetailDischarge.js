import React, { PureComponent } from 'react';
import FormSectionList from '../../../form-fields/FormSectionList';
import FormSection from '../../../form-fields/FormSection';

import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { valuesNames, valuesLabels } from '../forms.config';
import classNames from "classnames";

const DIAGNOSIS_TYPE = 'diagnoses';

export default class DocumentsDetailDischarge extends PureComponent {

  render() {
    let { detail, importHandler } = this.props;
    detail = detail || {};

    const date = getDDMMMYYYY(detail[valuesNames.DATE]);
    const dischargingDate = getDDMMMYYYY(detail[valuesNames.PI_DISCHARGING_DATE]);
    const addmissionDate = getDDMMMYYYY(detail[valuesNames.PI_ADMISSION_DATE]);

    return (
      <div className="form">
        <div className="form-group-wrapper">
          <div className="row-expand">
            {detail[valuesNames.NAME] ?
              <div className="col-expand-left">
                <div className="form-group">
                  <label className="control-label">{valuesLabels.NAME}</label>
                  <div className="form-control-static">{detail[valuesNames.NAME]}</div>
                </div>
              </div>
              : null
            }
            <div className={classNames({'col-expand-left': !detail[valuesNames.NAME], 'col-expand-right': detail[valuesNames.NAME]})}>
              <div className="form-group">
                <label className="control-label">{valuesLabels.DATE}</label>
                <div className="form-control-static">{date}</div>
              </div>
            </div>
          </div>

          <div className="row-expand">
            <div className="col-expand-left">
              <div className="form-group">
                <label className="control-label">{valuesLabels.AUTHOR}</label>
                <div className="form-control-static">{detail[valuesNames.AUTHOR]}</div>
              </div>
            </div>
          </div>

          <FormSectionList title={valuesLabels.TITLE_FACILITY}>
            <FormSection>
              <div className="form-group-wrapper">
                <div className="row-expand">
                  <div className="col-expand-left">
                    <div className="form-group">
                      <label className="control-label">{valuesLabels.FACILITY}</label>
                      <div className="form-control-static">{detail[valuesNames.FACILITY]}</div>
                    </div>
                  </div>
                </div>
              </div>
            </FormSection>
          </FormSectionList>

          <FormSectionList title={valuesLabels.TITLE_PATIENT_INDENTIFIER}>
            <FormSection>
              <div className="form-group-wrapper">
                <div className="row-expand">
                  { detail[valuesNames.PI_MRNTYPE] ?
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_MRNTYPE}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_MRNTYPE]}</div>
                      </div>
                    </div>
                    : null
                  }
                  { detail[valuesNames.PI_OTHTYPE] ?
                    <div className={classNames({'col-expand-left': !detail[valuesNames.PI_MRNTYPE], 'col-expand-right': detail[valuesNames.PI_MRNTYPE]})}>
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_OTHTYPE}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_OTHTYPE]}</div>
                      </div>
                    </div>
                    : null
                  }
                </div>

                <div className="row-expand">
                  { detail[valuesNames.PI_GMSTYPE] ?
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_GMSTYPE}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_GMSTYPE]}</div>
                      </div>
                    </div>
                    : null
                  }
                </div>

                <div className="row-expand">
                  { detail[valuesNames.PI_NAME] ?
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_NAME}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_NAME]}</div>
                      </div>
                    </div>
                    : null
                  }
                  { detail[valuesNames.PI_ID_TYPE] ?
                    <div className={classNames({'col-expand-left': !detail[valuesNames.PI_NAME], 'col-expand-right': detail[valuesNames.PI_NAME]})}>
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_ID_TYPE}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_ID_TYPE]}</div>
                      </div>
                    </div>
                    : null
                  }
                </div>

                <div className="row-expand">
                  { detail[valuesNames.PI_SOURCE] ?
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_SOURCE}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_SOURCE]}</div>
                      </div>
                    </div>
                    : null
                  }
                  { detail[valuesNames.PI_DISCHARGING] ?
                    <div className={classNames({'col-expand-left': !detail[valuesNames.PI_SOURCE], 'col-expand-right': detail[valuesNames.PI_SOURCE]})}>
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_DISCHARGING}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_DISCHARGING]}</div>
                      </div>
                    </div>
                    : null
                  }
                </div>

                <div className="row-expand">
                  { detail[valuesNames.PI_DISCHARGING_DATE] ?
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_DISCHARGING_DATE}</label>
                        <div className="form-control-static">{dischargingDate}</div>
                      </div>
                    </div>
                    : null
                  }
                </div>

                <div className="row-expand">
                  { detail[valuesNames.PI_CLINICAL] ?
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_CLINICAL}</label>
                        <div className="form-control-static">{detail[valuesNames.PI_CLINICAL]}</div>
                      </div>
                    </div>
                    : null
                  }
                  { detail[valuesNames.PI_ADMISSION_DATE] ?
                    <div className={classNames({'col-expand-left': !detail[valuesNames.PI_CLINICAL], 'col-expand-right': detail[valuesNames.PI_CLINICAL]})}>
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PI_ADMISSION_DATE}</label>
                        <div className="form-control-static">{addmissionDate}</div>
                      </div>
                    </div>
                    : null
                  }
                </div>
              </div>
            </FormSection>
          </FormSectionList>

          { detail[valuesNames.DIAGNOSIS] ?
            <FormSectionList title={valuesLabels.TITLE_DIAGNOSIS}>
              {detail[valuesNames.DIAGNOSIS].map((item, index) => {
                return (<FormSection
                  key={index}
                  title={`${valuesLabels.DG_PROBLEM}: ${item[valuesNames.DG_PROBLEM]}`}
                  isImportBtn
                  isAccordion
                  isBordered
                  onImportClick={importHandler(DIAGNOSIS_TYPE, item)}
                  theme='primary'
                >
                  <div className="form-group-wrapper">
                    <div className="row-expand">
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.DG_PROBLEM}</label>
                          <div className="form-control-static">{item[valuesNames.DG_PROBLEM]}</div>
                        </div>
                      </div>
                      <div className="col-expand-right">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.DG_DESCR}</label>
                          <div className="form-control-static">{item[valuesNames.DG_DESCR]}</div>
                        </div>
                      </div>
                    </div>
                    <div className="row-expand">
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.DG_TERMINOLOGY}</label>
                          <div className="form-control-static">{item[valuesNames.DG_TERMINOLOGY]}</div>
                        </div>
                      </div>
                      <div className="col-expand-right">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.DG_TERMINOLOGY_CODE}</label>
                          <div className="form-control-static">{item[valuesNames.DG_TERMINOLOGY_CODE]}</div>
                        </div>
                      </div>
                    </div>
                  </div>
                </FormSection>)
              })}
            </FormSectionList>
            : null
          }
        </div>
      </div>
    )
  }
}