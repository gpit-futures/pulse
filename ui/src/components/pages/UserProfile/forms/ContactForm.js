import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
