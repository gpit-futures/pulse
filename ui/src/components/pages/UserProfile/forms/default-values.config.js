import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
