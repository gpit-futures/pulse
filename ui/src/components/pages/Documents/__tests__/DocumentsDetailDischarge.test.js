import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import DocumentsDetailDischarge from '../DocumentsDetail/DocumentsDetailDischarge';
import { valuesNames, valuesLabels } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });
const DATE_TO_USE = new Date('2017');
const DATE_TO_USE_TEXT = '01-Jan-2017';
const DATE_TO_USE_TIME = DATE_TO_USE.getTime();
global.Date = jest.fn(() => DATE_TO_USE);
Date.now = jest.fn(() => DATE_TO_USE_TIME);

const testProps = {
  importHandler: () => {},
  detail: {
    [valuesNames.NAME]: 'NAME',
    [valuesNames.DATE]: 'DATE',
    [valuesNames.AUTHOR]: 'AUTHOR',
    [valuesNames.FACILITY]: 'FACILITY',
    [valuesNames.PI_MRNTYPE]: 'PI_MRNTYPE',
    [valuesNames.PI_OTHTYPE]: 'PI_OTHTYPE',
    [valuesNames.PI_GMSTYPE]: 'PI_GMSTYPE',
    [valuesNames.PI_NAME]: 'PI_NAME',
    [valuesNames.PI_ID_TYPE]: 'PI_ID_TYPE',
    [valuesNames.PI_SOURCE]: 'PI_SOURCE',
    [valuesNames.PI_DISCHARGING]: 'PI_DISCHARGING',
    [valuesNames.PI_DISCHARGING_DATE]: 'PI_DISCHARGING_DATE',
    [valuesNames.PI_CLINICAL]: 'PI_CLINICAL',
    [valuesNames.PI_ADMISSION_DATE]: 'PI_ADMISSION_DATE',
    [valuesNames.DIAGNOSIS]: [{
      [valuesNames.DG_PROBLEM]: 'DG_PROBLEM',
      [valuesNames.DG_DESCR]: 'DG_DESCR',
      [valuesNames.DG_TERMINOLOGY]: 'DG_TERMINOLOGY',
      [valuesNames.DG_TERMINOLOGY_CODE]: 'DG_TERMINOLOGY_CODE',
    }]
  }
};

describe('Component <DocumentsDetailDischarge />', () => {
	it('should renders with props correctly', () => {
		const component = shallow(<DocumentsDetailDischarge />);
    expect(component).toMatchSnapshot();

    expect(component.find('.form-group')).toHaveLength(3);
    expect(component.find('.form-group').at(0).find('.control-label').text()).toEqual(valuesLabels.DATE);
    expect(component.find('.form-group').at(1).find('.control-label').text()).toEqual(valuesLabels.AUTHOR);
    expect(component.find('.form-group').at(2).find('.control-label').text()).toEqual(valuesLabels.TYPE);

		component.setProps({ detail: testProps.detail, importHandler: testProps.importHandler });
		const diagnosisLength = testProps.detail[valuesNames.DIAGNOSIS].length;
    expect(component.find('FormSectionList')).toHaveLength(3);
    expect(component.find('FormSection')).toHaveLength(2 + diagnosisLength);

    expect(component.find('FormSectionList').at(0).props().title).toEqual(valuesLabels.TITLE_FACILITY);
    expect(component.find('FormSectionList').at(1).props().title).toEqual(valuesLabels.TITLE_PATIENT_INDENTIFIER);
    expect(component.find('FormSectionList').at(2).props().title).toEqual(valuesLabels.TITLE_DIAGNOSIS);

    expect(component.find('.form-group')).toHaveLength(14 + diagnosisLength * 4);

    const formControls = component.find('.form-group');
    expect(formControls.at(0).find('.control-label').text()).toEqual(valuesLabels.NAME);
    expect(formControls.at(1).find('.control-label').text()).toEqual(valuesLabels.DATE);
    expect(formControls.at(2).find('.control-label').text()).toEqual(valuesLabels.AUTHOR);
    expect(formControls.at(3).find('.control-label').text()).toEqual(valuesLabels.FACILITY);
    expect(formControls.at(4).find('.control-label').text()).toEqual(valuesLabels.PI_MRNTYPE);
    expect(formControls.at(5).find('.control-label').text()).toEqual(valuesLabels.PI_OTHTYPE);
    expect(formControls.at(6).find('.control-label').text()).toEqual(valuesLabels.PI_GMSTYPE);
    expect(formControls.at(7).find('.control-label').text()).toEqual(valuesLabels.PI_NAME);
    expect(formControls.at(8).find('.control-label').text()).toEqual(valuesLabels.PI_ID_TYPE);
    expect(formControls.at(9).find('.control-label').text()).toEqual(valuesLabels.PI_SOURCE);
    expect(formControls.at(10).find('.control-label').text()).toEqual(valuesLabels.PI_DISCHARGING);
    expect(formControls.at(11).find('.control-label').text()).toEqual(valuesLabels.PI_DISCHARGING_DATE);
    expect(formControls.at(12).find('.control-label').text()).toEqual(valuesLabels.PI_CLINICAL);
    expect(formControls.at(13).find('.control-label').text()).toEqual(valuesLabels.PI_ADMISSION_DATE);

    expect(formControls.at(0).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.NAME]);
    expect(formControls.at(1).find('.form-control-static').text()).toEqual(DATE_TO_USE_TEXT);
    expect(formControls.at(2).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.AUTHOR]);
    expect(formControls.at(3).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.FACILITY]);
    expect(formControls.at(4).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_MRNTYPE]);
    expect(formControls.at(5).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_OTHTYPE]);
    expect(formControls.at(6).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_GMSTYPE]);
    expect(formControls.at(7).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_NAME]);
    expect(formControls.at(8).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_ID_TYPE]);
    expect(formControls.at(9).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_SOURCE]);
    expect(formControls.at(10).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_DISCHARGING]);
    expect(formControls.at(11).find('.form-control-static').text()).toEqual(DATE_TO_USE_TEXT);
    expect(formControls.at(12).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.PI_CLINICAL]);
    expect(formControls.at(13).find('.form-control-static').text()).toEqual(DATE_TO_USE_TEXT);

    // Test Diagnosis Section
    const diagnosisSection = component.find('FormSectionList').at(2).find('FormSection').at(0);
    const diagnose = testProps.detail[valuesNames.DIAGNOSIS][0];

    expect(diagnosisSection.find('.control-label').at(0).text()).toEqual(valuesLabels.DG_PROBLEM);
    expect(diagnosisSection.find('.control-label').at(1).text()).toEqual(valuesLabels.DG_DESCR);
    expect(diagnosisSection.find('.control-label').at(2).text()).toEqual(valuesLabels.DG_TERMINOLOGY);
    expect(diagnosisSection.find('.control-label').at(3).text()).toEqual(valuesLabels.DG_TERMINOLOGY_CODE);

    expect(diagnosisSection.find('.form-control-static').at(0).text()).toEqual(diagnose[valuesNames.DG_PROBLEM]);
    expect(diagnosisSection.find('.form-control-static').at(1).text()).toEqual(diagnose[valuesNames.DG_DESCR]);
    expect(diagnosisSection.find('.form-control-static').at(2).text()).toEqual(diagnose[valuesNames.DG_TERMINOLOGY]);
    expect(diagnosisSection.find('.form-control-static').at(3).text()).toEqual(diagnose[valuesNames.DG_TERMINOLOGY_CODE]);

		expect(component).toMatchSnapshot();
	});
});

