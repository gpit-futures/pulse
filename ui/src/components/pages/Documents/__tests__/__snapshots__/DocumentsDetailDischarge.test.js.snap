// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <DocumentsDetailDischarge /> should renders with props correctly 1`] = `
<div
  className="form"
>
  <div
    className="form-group-wrapper"
  >
    <div
      className="row-expand"
    >
      <div
        className="col-expand-left"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Document Date
          </label>
          <div
            className="form-control-static"
          >
            01-Jan-2017
          </div>
        </div>
      </div>
    </div>
    <div
      className="row-expand"
    >
      <div
        className="col-expand-left"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Author
          </label>
          <div
            className="form-control-static"
          />
        </div>
      </div>
    </div>
    <FormSectionList
      title="Facility"
    >
      <FormSection
        isAccordion={false}
        isBordered={false}
        isImportBtn={false}
      >
        <div
          className="form-group-wrapper"
        >
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Name
                </label>
                <div
                  className="form-control-static"
                />
              </div>
            </div>
          </div>
        </div>
      </FormSection>
    </FormSectionList>
    <FormSectionList
      title="Patient Identifier"
    >
      <FormSection
        isAccordion={false}
        isBordered={false}
        isImportBtn={false}
      >
        <div
          className="form-group-wrapper"
        >
          <div
            className="row-expand"
          />
          <div
            className="row-expand"
          />
          <div
            className="row-expand"
          />
          <div
            className="row-expand"
          />
          <div
            className="row-expand"
          />
          <div
            className="row-expand"
          />
        </div>
      </FormSection>
    </FormSectionList>
  </div>
</div>
`;

exports[`Component <DocumentsDetailDischarge /> should renders with props correctly 2`] = `
<div
  className="form"
>
  <div
    className="form-group-wrapper"
  >
    <div
      className="row-expand"
    >
      <div
        className="col-expand-left"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Document Name
          </label>
          <div
            className="form-control-static"
          >
            NAME
          </div>
        </div>
      </div>
      <div
        className="col-expand-right"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Document Date
          </label>
          <div
            className="form-control-static"
          >
            01-Jan-2017
          </div>
        </div>
      </div>
    </div>
    <div
      className="row-expand"
    >
      <div
        className="col-expand-left"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Author
          </label>
          <div
            className="form-control-static"
          >
            AUTHOR
          </div>
        </div>
      </div>
    </div>
    <FormSectionList
      title="Facility"
    >
      <FormSection
        isAccordion={false}
        isBordered={false}
        isImportBtn={false}
      >
        <div
          className="form-group-wrapper"
        >
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Name
                </label>
                <div
                  className="form-control-static"
                >
                  FACILITY
                </div>
              </div>
            </div>
          </div>
        </div>
      </FormSection>
    </FormSectionList>
    <FormSectionList
      title="Patient Identifier"
    >
      <FormSection
        isAccordion={false}
        isBordered={false}
        isImportBtn={false}
      >
        <div
          className="form-group-wrapper"
        >
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  MRM Type
                </label>
                <div
                  className="form-control-static"
                >
                  PI_MRNTYPE
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  OTH Type
                </label>
                <div
                  className="form-control-static"
                >
                  PI_OTHTYPE
                </div>
              </div>
            </div>
          </div>
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  GMS Type
                </label>
                <div
                  className="form-control-static"
                >
                  PI_GMSTYPE
                </div>
              </div>
            </div>
          </div>
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Responsible Professional
                </label>
                <div
                  className="form-control-static"
                >
                  PI_NAME
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Responsible Professional Type
                </label>
                <div
                  className="form-control-static"
                >
                  PI_ID_TYPE
                </div>
              </div>
            </div>
          </div>
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  From
                </label>
                <div
                  className="form-control-static"
                >
                  PI_SOURCE
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Discharging Organisation
                </label>
                <div
                  className="form-control-static"
                >
                  PI_DISCHARGING
                </div>
              </div>
            </div>
          </div>
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Discharging Date
                </label>
                <div
                  className="form-control-static"
                >
                  01-Jan-2017
                </div>
              </div>
            </div>
          </div>
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Clinical Synopsis
                </label>
                <div
                  className="form-control-static"
                >
                  PI_CLINICAL
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Date of Admission
                </label>
                <div
                  className="form-control-static"
                >
                  01-Jan-2017
                </div>
              </div>
            </div>
          </div>
        </div>
      </FormSection>
    </FormSectionList>
    <FormSectionList
      title="Diagnosis List"
    >
      <FormSection
        isAccordion={true}
        isBordered={true}
        isImportBtn={true}
        key="0"
        theme="primary"
        title="Problem: DG_PROBLEM"
      >
        <div
          className="form-group-wrapper"
        >
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Problem
                </label>
                <div
                  className="form-control-static"
                >
                  DG_PROBLEM
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Description
                </label>
                <div
                  className="form-control-static"
                >
                  DG_DESCR
                </div>
              </div>
            </div>
          </div>
          <div
            className="row-expand"
          >
            <div
              className="col-expand-left"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Terminology
                </label>
                <div
                  className="form-control-static"
                >
                  DG_TERMINOLOGY
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  Terminology Code
                </label>
                <div
                  className="form-control-static"
                >
                  DG_TERMINOLOGY_CODE
                </div>
              </div>
            </div>
          </div>
        </div>
      </FormSection>
    </FormSectionList>
  </div>
</div>
`;
